app/src/main/java/com/example/ar_reshare/SignUpFragment.java

```diff
@@ -2,53 +2,106 @@
 import android.graphics.Color;
 import android.icu.util.Calendar;
+import android.net.Uri;
 import android.os.Bundle;
 import android.os.CountDownTimer;
+import android.os.Environment;
+import android.provider.MediaStore;
 import android.text.Editable;
 import android.text.TextWatcher;
+import android.util.Pair;
+import android.view.Gravity;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.Button;
 import android.widget.DatePicker;
 import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.PopupWindow;
 import android.widget.TextView;
 import android.widget.Toast;
 
+import androidx.activity.result.ActivityResult;
+import androidx.activity.result.ActivityResultCallback;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
 import androidx.annotation.Nullable;
+import androidx.core.content.FileProvider;
 import androidx.fragment.app.Fragment;
 
+import java.io.File;
+import java.io.IOException;
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
 import java.util.Arrays;
 import java.util.Date;
 import java.util.GregorianCalendar;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.concurrent.TimeUnit;
 
 public class SignUpFragment extends Fragment {
 
+    private final int DIALOG_TIME = 3000; // milliseconds
     private final int GREEN_COLOUR = Color.parseColor("#32a852");
+    private final int GREEN_BACKGROUND_COLOR = Color.parseColor("#198235");
     private final int RED_COLOUR = Color.parseColor("#ab2a1f");
     private final int DEFAULT_TEXT_COLOUR = Color.parseColor("#363636");
+    private final int BLUE_BACKGROUND_COLOUR = Color.parseColor("#4C62DC");
     private final int CLICKABLE_COLOUR = Color.parseColor("#4C62DC");
     private final int NOT_CLICKABLE_COLOUR = Color.parseColor("#7080db");
 
     private final List<Character> SPECIAL_CHARACTERS = Arrays.asList('@', '!', '?', '%', '+', '-', '\\', '/', '\'', '$', '#', '^', ':', ';', '(', ')', '[', ']', '{', '}', '~', '_', '.');
 
     private final long MINIMUM_AGE_REQUIRED = 18L;
 
+    private File profilePicture = null;
+    private Uri profilePictureURI = null;
+
+    private View addPhotoWindow;
+    private ActivityResultLauncher cameraLauncher;
+    private ActivityResultLauncher galleryLauncher;
+
     public SignUpFragment() {
         // Required empty public constructor
     }
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
+        cameraLauncher = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), new ActivityResultCallback<ActivityResult>() {
+            @Override
+            public void onActivityResult(ActivityResult result) {
+                if (result.getResultCode() == -1) {
```

Comment:
All amazing work but maybe you could change -1 to RESULT_OK for better code readability

Reply:
That is a good idea! I have changed it with a new commit